app/controllers/api/service_templates_controller.rb

```diff
@@ -4,5 +4,56 @@ class ServiceTemplatesController < BaseController
     include Subcollections::Tags
     include Subcollections::ResourceActions
     include Subcollections::ServiceRequests
+
+    def create_resource(_type, _id, data)
+      # Temporarily only supporting atomic.
+      # Will update API to support composite separately.
+      raise 'Service Type composite not supported' if data['service_type'] == 'composite'
+      create_atomic(data).tap(&:save!)
+    rescue => err
+      raise BadRequestError, "Could not create Service Template - #{err}"
+    end
+
+    private
+
+    def create_atomic(data)
+      validate_atomic_data(data)
+      service_template = ServiceTemplate.new(data.except('request_info'))
+      service_template.add_resource(create_service_template_request(data))
+      set_new_resource_actions(data['request_info'], service_template)
+    end
+
+    def validate_atomic_data(data)
+      raise 'Must provide request info' unless data.key?('request_info')
+      raise 'Provisioning Entry Point is required' unless data['request_info']['fqname']
```

Comment:
Do we really want to enforce an provisioning entry point? I think now the backend have default value for all provisioning types.
In general I don't know whether it is necessary to add validation for too many fields, because even if some fields are missing at creation time, they can be added through update API.